File: permissionmadeeasy/src/main/java/com/somesh/permissionmadeeasy/intefaces/PermissionListener.java
package com.somesh.permissionmadeeasy.intefaces;

import java.util.ArrayList;

/**
 * If this code works, it was written by Somesh Kumar  on 28 November, 2017. If not, I don't know who wrote it.
 */
public interface PermissionListener {
    void onPermissionsGranted(int requestCode, ArrayList<String> acceptedPermissionList);

    void onPermissionsDenied(int requestCode, ArrayList<String> deniedPermissionList);
}

File: permissionmadeeasy/src/main/java/com/somesh/permissionmadeeasy/intefaces/PermissionListener.kt
package com.somesh.permissionmadeeasy.intefaces

import java.util.*

/**
 * If this code works, it was written by Somesh Kumar  on 28 November, 2017. If not, I don't know who wrote it.
 */
interface PermissionListener {
    fun onPermissionsGranted(requestCode: Int, acceptedPermissionList: ArrayList<String>)
    fun onPermissionsDenied(requestCode: Int, deniedPermissionList: ArrayList<String>)
}